File: shustagram-web/shustagram/api/models/Post.js
module.exports = {

    customToJSON: function() {
        const moment = sails.moment
        this.fromNow = moment(this.createdAt).fromNow()
        return this
    },

    attributes: {
        text: {
            type: 'string', required: true
        },

        imageUrl: {
            type: 'string',
            defaultsTo: ''
        },

        user: {
            model: 'user'
        },

        numLikes: {
            type: 'number', defaultsTo: 0
        },
    }
}

File: shustagram-ios/shustagram/shustagram/Model/Post.swift
//
//  Post.swift
//  shustagram
//
//  Created by 徐乾智 on 7/22/21.
//

import Foundation

struct Post: Decodable {
    let id: String
    let text: String
    let createdAt: Int
    let user: User
    let imageUrl: String
    
    var fromNow: String?
    var comments: [Comment]?
    
    var hasLiked: Bool?
    
    var numLikes: Int
}
